src/main/scala/firrtl/passes/memlib/ToMemIR.scala

```diff
@@ -11,14 +11,15 @@ import firrtl.ir._
   *   - read latency and write latency of one
   *   - only one readwrite port or write port
   *   - zero or one read port
+  *   - undefined read-under-write behavior
   */
 object ToMemIR extends Pass {
   /** Only annotate memories that are candidates for memory macro replacements
-    * i.e. rw, w + r (read, write 1 cycle delay)
+    * i.e. rw, w + r (read, write 1 cycle delay) and read-under-write "undefined."
```

Comment:
Are there no memory compilers that support a defined behavior for this case? I could see a memory compiler generating logic to make this case defined, but I am lacking experience with memory compilers here.

Reply:
Currently we lack the ability to enforce any downstream behavior, so Jack requested I put this in as an extra safety check. In practice, memory IPs would implement one of the three options or expose it as a flag.